solutions/USACO dec20-cowntagion.mdx

```diff
@@ -2,14 +2,69 @@
 id: usaco-1062
 source: USACO Silver 2020 December
 title: Cowntagion
-author: Kevin Sheng
+author: Kevin Sheng (Java, Python), Tanish Tyagi (C++)
 ---
 
 [Official Editorial](http://www.usaco.org/current/data/sol_prob1_silver_dec20.html)
 
 <LanguageSection>
 
+<CPPSection>
+
+## C++ Implementation
+
+```cpp 
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define maxn 100005
+int n;
+vector<int> adj[maxn];
+
+int dfs(int start, int parent) {
+	int ans = 0;
+	int cows = adj[start].size();
+	if (parent == -1) cows++;
+
+	int currCows = 1;
+	int days = 0;
+	//count the number of superspreader events needed
+	while (currCows < cows) {
+		days++;
+		currCows *= 2;
+	}
+	ans += days;
+
```

Comment:
// Count the number of roads each cow has to travel.

Reply:
I don't think this is what the code is doing? I think it should probably be `// Send one cow to each adjacent farm without a sick cow`.

Reply:
Well this is what it calculates...

Reply:
I think it's not really about each cow, but a cumulative count of roads traveled over all the cows?

Reply:
Oh true. Please change to: 

`// Count the number of roads traveled by all the cows`

Reply:
Actually, it also includes the superspreader events that happen at the other farms, so saying `the number of roads` seems a little inaccurate. Maybe just saying that it is a dfs with `// Send one cow to each adjacent farm without a sick cow` should be good? You can change the wording if you want.

Reply:
:/ isn't `days` is the number of superspreader events it takes to make the # of sick cows at `start` at least `cows`?

Reply:
this is about the section below (dfs for each of the adjacent farms)